### ISSUE TITLE
🆕 Software Suggestion | Cyber: A decentralized search engine  

### ISSUE THREAD

**Issue by serejandmyself** [privacytoolsIO/privacytools.io#1646]
## TL;DR
I am affiliated with the project directly. More so, I am the ecosystem development lead of the project. I have opened a PR previously, but according to a few users, it seems that I have opened it without following some of the rules of suggesting software. I have decided to close the PR in favour of this issue first hand, and re-open it if it will be appropriate. You can find the PR here: https://github.com/privacytoolsIO/privacytools.io/pull/1645

## Basic Information
Cyber is a decentralized Google, for relevant and provable answers. It is a blockchain based search protocol, with a dynamic ranking mechanism, spam protection mechanisms and economic incentives. It is build with the help of IPFS and Tendermint (used for consensus). 

**Name:** Cyber
**Category:** Search-engines
**URL(s):** 
GitGub: https://github.com/cybercongress/congress
Congress website: https://cybercongress.ai/
WP: https://ipfs.io/ipfs/QmceNpj6HfS81PcCaQXrFMQf7LR5FTLkdG9sbSRNy3UXoZ
ElI-5 FAQ: https://github.com/cybercongress/congress/blob/master/ecosystem/ELI-5%20FAQ.md
Alpha search page: https://cyber.page/

## Description
Cyber is essentially creating a brand new protocol for adding and searching information onto a knowledge graph (a compilation of facts about something that provides meaning to the user). And ranks this information (cyber uses the original PageRank mechanism with a spam protection mechanism and a few other dynamic properties. It should be noted that the rank mechanism, can be changed via governance).

Different types of users create links between 2 IPFS hashes (a peer-to-peer protocol, that is designed to make the web faster, safer, and more open and place them on a knowledge graph by spending bandwidth (amount of data that can be transmitted over a fixed period of time).

That content is then dynamically ranked with the help of digital tokens and the current parameters of the networks load. This makes the rank dynamic.

All this is computed by validators (a program or a computer that are responsible for checking the validity of something). The validators do so, by using their GPU's.

This allows to search data on the web, rank it, query it and create knowledge databases without blackbox intermediaries.

The interesting thing is that such a simple mechanism allows creating a lot of powerful tools as a result. For example: unified semantics, SEO instruments, autonomous robots, and a lot more.

Cyber is NOT pure P2P, as blockchain acts as an intermediary between users. Cyber is based on economic incentives and open-source technologies (everything is open-source). Cyber is not the actual browser, it is a protocol. The browser (Cyb) is a personal application that runs on top of the protocol and is private to each user.

As of today cyber does NOT have private transactions by design (although that is something cyber would like to achieve). This means that each transaction made (linking including) is represented by a hash. This hash includes info about what was linked, by which node and a timestamp. This allows cyber to check the validity of each link in the network and its rank. Making sure that any content linked, can actually be traced to the source (via the use of merkle trees).

**Comment by Mikaela** [comment]
Continuing from #1645 

I apologise, I am not currently my best possible self.

> Hey! By no means I am hiding the fact I am affiliated with the project. Of course I am. It's all over my profile. I wasn't aware of the fact that this is against the rules. Seems like a lot of politics to open a PR =)

I wish you would have brought it up in the issue and PR description as I have a feeling that I am currently the only one going through issues and labeling them and in my current situation I am barely checking the homepages of the suggested projects without even thinking about profiles. I hope the rest of @privacytoolsIO/editorial recovers from holidays soon.

> TBH, I still do not understand the PTIO thing above.

https://www.reddit.com/r/privacytoolsIO/comments/dza44p/who_owns_your_favorite_privacy_service_part_2_the/ has a list of questions we would hope to hear your answer to, preferably to #1646, I guess.

> I guess I will just close the PR for now and add info to the issue (although after reading the contributing guidelines, I did not see the inability to open a PR along with an issue. In fact it seems more logical to me).

I hope #1651 clarifies that. I do like there being an issue and a PR to close the issue at the same time, while the PR may take a long time to get properly looked at.

**Comment by serejandmyself** [comment]
> Continuing from #1645
> 
> I apologise, I am not currently my best possible self.
> 
No need to apologize, I was the one who didn't put the proper description to start with. I hope it's more sufficient now. Please let me know if I need to add anything else to it! 
> https://www.reddit.com/r/privacytoolsIO/comments/dza44p/who_owns_your_favorite_privacy_service_part_2_the/ has a list of questions we would hope to hear your answer to, preferably to #1646, I guess.
> 
Ok, get it now. For some reason I kept not seeing the questions. I have added answers to #1646. Thanks for taking the time to clarify

**Comment by serejandmyself** [comment]
> Who owns the company/organization? What percentage does each owner hold? (December 31 of prior year and current date)

There is no company/organisation at all. We do not believe in legislation. We do not believe in law by default. There is a DAO (a decentralized autonomous organisation on the blockchain, run with the help of smart contracts) that is called cyber~ Congress. For launching the protocol and for financing its first 3 years of work (roughly 1.2 million USD - disclosed in our FAQ's), it receives up to 10% of THC tokens, which it will stake (on the same "rules" as the other participants) and receive an equal amount of CYB tokens. 
The funding of the protocol is the following: the first round of donations goes to cyber congress, which will be staked for a validator, and the rewards will be used by cyber~Congress to fund proposals from team members (for example one of those is the ecosystem fund to help evolve the protocol). 
The main round of donations will be governed by the community itself, which is going to be held in ETH, thorough governance proposals. 
There was also an initial donation round, in which 10 people participated (although it was open to the broad public), for which they will receive up to 10% of CYB tokens too. 
All this info is public on our GitHub, in our White paper and our economic docs. By our calculations, the congress and affiliated people will have up to 14% of the tokens. We understand that this is quite a lot. And our goal is to wash down to 2% in the next 15 - 20 years (of course I am referring to the tokens of the congress. We cannot washdown the tokens of non-congress members, even if they are affiliated).
In any case, the whole protocol is community-governed. It should also be noted that 10% were gifted to the following communities: 8% to over 1 million ETH addresses, 1% to ATOM addresses and 1% to Urbit addresses. 

> Have you changed how information is processed and shared in the last year?

No, we use GitHub for ALL of our information. It is our goal to open-source ALL the information that we produce. 

The blog, website, etc are all run from our GitHub account. There is, of course, Twitter, reddit, etc which aren't. But they are all public. 

>Do you share data -- even "fuzzed" or "anonymized" data -- with any of the owners/shareholders or any other company or organization server?

We do not collect any data. The blockchain is, of course, open to the public. All transactions are hashed of course.
 
> Please share a diagram showing how information flows when a user interacts with your service.

I believe the best place for this is here: https://ipfs.io/ipfs/QmceNpj6HfS81PcCaQXrFMQf7LR5FTLkdG9sbSRNy3UXoZ
Please see pages: 2,3,4,8 and 9 for such diagrams 

> Which components of your service are not open source? Where can we find the code for the open source components?

They are ALL open-source. Including our plans, writings, code, etc, etc, etc
Our GitHub account: https://github.com/cybercongress
 
> Have you had any independent audits in the last three years? Please share the dates of those audits and audit reports.

No. We have not done any audits of the code yet. It is open to the public. 

> If you require sign-up or account creation, do consumers have easy access to tools to delete their data? Can they delete everything on the servers or just the local cache?

There is no sign-up as such. An identity is a pair of keys created on the blockchain. 

> Is there a way for consumers to view any information you have collected about them?

We do not collect consumer information. Any user can open the explorer to see available info about any transaction, account, etc 

> What is your business model? How do you fund operations and make money?

As of now, the project is self-funded. In the future, it is planned to fund from the staking rewards of ATOMs 

> Do you offer a transparency report? Warrant canary?|

No. The project is open-sourced. 100%

> How is data secured (in transit and at rest)?

It is secured by the blockchain. The more nodes that will be connected to the protocol, the better security it will have. The more tokens will be staked, the more secure the chain will become. 

> Who has access to customer data?

Anyone with access to the blockchain / explorer. But there is no customer data as such to see there. 

> What 3rd parties have access to customer data?

None

> What processes do you have in place if there is unauthorized access to data?

This means the blockchain has been hacked. As of today, this has never happened. If such a case arises, it will mean a lot of changes in cryptography generally. But in terms of a solution, I believe a fork is possible. Bu this will be decided by a governance protocol. 

> What customer data is collected, how often, and in what level of identification?

None, in terms of "data collection". BUT,
Each cyberlink contains hashes of what was linked, when, with what weight and by which node. 
When Cyb, the browser is out and stable, I assume that some people would sell their own data, via APIs. We cannot control this. The browser in our case will be a personal application on top of the protocol that the user and only the user controls. 

>  Will changes to your Terms & Conditions and privacy policies be communicated to end users at least 30 days in advance of any changes? How will these changes be communicated?

We do not have terms & conditions as of now. The license is: don't believe, don't fear, don't ask

> Do you commit to notifying consumers at least 30 days in advance if you plan to sell your service or take on new investors/owners? How will this be communicated?

After the donations and governance contracts, the protocol is solely community governed. It will be up to the protocol to decide on such things, how to communicate this, etc. 

All the information provided is true and correct to today. Once again, the idea of the protocol is to be community governed via a set of smart contracts and economy. The governing protocol may change pretty much anything that is parametrized (which is almost everything...) though voting mechanisms and A/B testing

**Comment by lrq3000** [comment]
@serejandmyself very interesting project, there would be at least one great privacy advantage in that there would be no (or at least a network community agreed) filtering bubble, so some form of censorship resistance. And when requests will be private by default, it will be even better.

Anyway I tried [searching](https://cyber.page/#/) for some common words such as "cats" or "dogs" and had no results. Is the search engine backend down at the moment?

Also may I suggest to provide a more visible theme, because with my blue light filters, it's impossible to see where the search text box is (and also it's weird that it moves from the middle to the top of the window without any indication or button to show that it's a search box). I would suggest to provide a light background by default, or make the box outline more visible (choose a different color) and add a search button/icon at the right to show that it's a search box.

**Comment by serejandmyself** [comment]
Hey @lrq3000 

Thanks for the comments. You are right, the first privacy focus is the lack of a blackbox intermediary. The second is the hashing of the search queries. The third is the fact that since CYB (the actual browser app - not mentioned here), is, in fact, a personal application on top of the protocol, unique for each person, which works on the machine of the person. Hence it doesn't share/collect any info (unless the user concisely connects his own API's to sell/trade data). Of course, it shares the hashed info on what, by whom and when was cyberlinked. 

The fourth, future, privacy point. Is that we believe that private tx should be eventually implemented by design into the protocol. Right now we are not concentrated on it, as the info is still hashed and the work on the protocol is more important. But, in the future (not less than the privacy tech in blockchain will be ready, so we are speaking a few years), the tx will become completely private (meaning there will even be no hashes of info), even though, stay auditable and verifiable.

As for the search engine. I'm glad you checked it out. Please keep in mind that it's still a public testnet. The network is very small and it learns as it goes. This means that someone has to actually semantically link the words `dog` or `cat` for them to appear in the search and ranking. Think of it as web3 being a pretty much of an empty box, which waits eagerly for new info. We are working on crawlers which will help users to index the old web, into the new web, firstly such sites as Github, Wikipedia, internet archive machine, etc

Please check [this guide](https://discuss.ipfs.io/t/cyber-page-inter-planetary-search-app-for-web3/7049) on how to use cyber.page to make sure it's working =)

Thanks for the feedback on the search. We are eagerly collecting user feedback to improve it. Much appreciated!

**Comment by 5a384507-18ce-417c-bb55-d4dfcc8883fe** [comment]
Hello, would you consider changing you Google analytics and fonts for a FLOSS alternative? I think Matomo for analytics and Fork Awesome for fonts could work.

**Comment by serejandmyself** [comment]
> Hello, would you consider changing you Google analytics and fonts for a FLOSS alternative? I think Matomo for analytics and Fork Awesome for fonts could work.

Hey, why did you decide that we use google analytics?

**Comment by 5a384507-18ce-417c-bb55-d4dfcc8883fe** [comment]
> Hey, why did you decide that we use google analytics?

I went to your website and both uMatrix and uBlock Origin tell me that you are using them.

**Comment by serejandmyself** [comment]
> > Hey, why did you decide that we use google analytics?
> 
> I went to your website and both uMatrix and uBlock Origin tell me that you are using them.

I assume you are referring to the DAO website (cybercongress.ai) not the protocol website - cyber.page

The DAO is simply the project behind launching the protocol. That's not the product page. There are no trackers on cyber.page

But good point. I will bring it up with the team

**Comment by 5a384507-18ce-417c-bb55-d4dfcc8883fe** [comment]
> > > Hey, why did you decide that we use google analytics?
> > 
> > 
> > I went to your website and both uMatrix and uBlock Origin tell me that you are using them.
> 
> I assume you are referring to the DAO website (cybercongress.ai) not the protocol website - cyber.page
> 
> The DAO is simply the project behind launching the protocol. That's not the product page. There are no trackers on cyber.page
> 
> But good point. I will bring it up with the team

I still see G fonts on it but it is better than G analytics. Thanks for taking the time to go through this!

**Comment by serejandmyself** [comment]
> > > > Hey, why did you decide that we use google analytics?
> > > 
> > > 
> > > I went to your website and both uMatrix and uBlock Origin tell me that you are using them.
> > 
> > 
> > I assume you are referring to the DAO website (cybercongress.ai) not the protocol website - cyber.page
> > The DAO is simply the project behind launching the protocol. That's not the product page. There are no trackers on cyber.page
> > But good point. I will bring it up with the team
> 
> I still see G fonts on it but it is better than G analytics. Thanks for taking the time to go through this!

We will see whats up with GA on the DAO site. 

You can check out this doc (in the making) for [references](https://github.com/cybercongress/congress/blob/master/ecosystem/Cyber%20Homestead%20doc.md)

**Comment by nitrohorse** [comment]
Hi @serejandmyself, appreciate you answering the [list of questions](https://github.com/privacytoolsIO/privacytools.io/issues/1846) and for being responsive to issues like [this one](https://github.com/cybercongress/congress/issues/318). The "questions list" is a work-in-progress still and not yet part of our review process, so apologies for the lack of clarity that answering them isn't a guarentee to be listed on PrivacyTools.

As [you mentioned](https://github.com/privacytoolsIO/privacytools.io/issues/1646#issuecomment-581144595):

>  Please keep in mind that it's still a public testnet. The network is very small and it learns as it goes.

I think Cyber isn't at a stage yet to be considered for addition to PrivacyTools -- the project still feels immature. It's difficult to take seriously, and I'm apprehensive of "blockchain"-based products ([1](https://cybercongress.ai/takeoff-proposal/), [2](https://github.com/cybercongress/go-cyber/blob/master/README.md#the-end-of-google), [3](https://github.com/cybercongress/go-cyber/blob/master/LICENSE)).

Thanks again for reaching out though, best of luck with the project. As the project grows and matures more, feel free to open a new issue.
